crates/proto/go/src/resolve.rs

```diff
@@ -0,0 +1,115 @@
+use crate::GoLanguage;
+use core::str;
+use lenient_semver::Version;
+use log::debug;
+use proto_core::{
+    async_trait, Describable, ProtoError, Resolvable, VersionManifest, VersionManifestEntry,
+};
+use std::collections::BTreeMap;
+use std::process::Command;
+
+trait BaseVersion {
+    fn base_version(&self) -> String;
+}
+
+impl<'a> BaseVersion for Version<'a> {
+    fn base_version(&self) -> String {
+        format!("{}.{}", self.major, self.minor)
+    }
+}
+
+#[async_trait]
+impl Resolvable<'_> for GoLanguage {
+    fn get_resolved_version(&self) -> &str {
+        self.version.as_ref().unwrap()
+    }
+
+    async fn load_manifest(&self) -> Result<VersionManifest, ProtoError> {
+        let mut alias_max = BTreeMap::new();
+        let mut aliases = BTreeMap::new();
+        let mut versions = BTreeMap::new();
+
+        let output = Command::new("git")
+            .args(["ls-remote", "--tags", "https://github.com/golang/go/"])
+            .output()
+            .expect("failed to execute process");
```

Comment:
expects will panic the program, convert them to ProtoError

Reply:
updated